panel/reactive.py

```diff
@@ -317,7 +317,8 @@ def _scheduled_update_model(
         try:
             self._update_model(events, msg, root, model, doc, comm)
         finally:
-            del self._in_process__events[doc]
+            if doc in self._in_process__events:
+                del self._in_process__events[doc]
```

Comment:
```suggestion
            self._in_process__events.pop(doc, None)
```

This should be more threadsafe